CMSgov/design-system
# [WNMGDS-2074] Release script updates

## Summary

1. [WNMGDS-2074](https://jira.cms.gov/browse/WNMGDS-2074)
2. Split our release script into 2 new scripts `bump-and-tag-release.sh` and `bump-main.sh` as these two processes are separate from each other. By doing our version bumps twice instead of cherry-picking a commit from one place to another, we avoid any possible errors with branch collisions or conflict resolution. Overall I believe this is a faster process as a result.
3. The `bump-and-tag-release.sh` script gets run as `yarn release` by default, to be run in a branch to tag and bump, then push tags. That is all that's needed to kick off a release with jenkins.
4. The `bump-main.sh` script gets run with `yarn release:bump` (maybe just `yarn bump`?) and creates a new branch off main, bumps versions without tagging and automagically creates a bump PR from that branch into main with `gh`

- I'm assuming we all have gh installed on our machines and that is a part of our tooling for the team. If not then it should be!
- [ ] Need to update [confluence documentation](https://confluence.cms.gov/display/HCDSG/Creating+a+code+release)

## How to test

1. `yarn release` in this branch, bump versions to whatever, see that it runs then run `yarn release -u` to undo those changes.
2. `yarn release:bump` in this branch, bump versions to whatever, see that PR is created, close PR and delete branch.

## Checklist
